ui-admin/src/study/surveys/AnswerMappingEditor.tsx

```diff
@@ -0,0 +1,305 @@
+import { Modal, ModalBody, ModalFooter, ModalHeader, ModalTitle } from 'react-bootstrap'
+import { OnChangeAnswerMappings } from '../../forms/formEditorTypes'
+
+
+const AnswerMappingLabels: {[key in AnswerMappingTargetType]: string} = {
+  'PROXY': 'Proxy',
+  'PROFILE': 'Profile',
+  'PROXY_PROFILE': 'Proxy Profile'
+}
+
+const ProfileFields: { [key: string]: AnswerMappingMapType } = {
+  'givenName': 'STRING_TO_STRING',
+  'familyName': 'STRING_TO_STRING',
+  'contactEmail': 'STRING_TO_STRING',
+  'phoneNumber': 'STRING_TO_STRING',
+  'birthDate': 'STRING_TO_LOCAL_DATE',
+  'doNotEmail': 'STRING_TO_BOOLEAN',
+  'doNotEmailSolicit': 'STRING_TO_BOOLEAN',
+  'sexAtBirth': 'STRING_TO_STRING',
+  'mailingAddress.street1': 'STRING_TO_STRING',
+  'mailingAddress.street2': 'STRING_TO_STRING',
+  'mailingAddress.city': 'STRING_TO_STRING',
+  'mailingAddress.state': 'STRING_TO_STRING',
+  'mailingAddress.postalCode': 'STRING_TO_STRING',
+  'mailingAddress.country': 'STRING_TO_STRING'
+}
+
+const ProxyFields: { [key: string]: AnswerMappingMapType } = {
+  'isProxy': 'STRING_TO_BOOLEAN'
+}
+
+const AnswerMappingTargets = {
+  'PROFILE': ProfileFields,
+  'PROXY': ProxyFields,
+  'PROXY_PROFILE': ProfileFields
+}
+
+const AnswerMappingMapTypeLabels: { [key in AnswerMappingMapType]: string} = {
+  'STRING_TO_STRING': 'Text',
+  'STRING_TO_BOOLEAN': 'Boolean',
+  'STRING_TO_LOCAL_DATE': 'Date'
+}
+
+/**
+ * Table which allows viewing, deleting, and creating new answer mappings.
+ */
+export default function AnswerMappingEditor(
+  {
+    formContent, initialAnswerMappings, onChange
+  } : {
+    formContent: FormContent, initialAnswerMappings: AnswerMapping[], onChange: OnChangeAnswerMappings
+  }
+) {
+  const [answerMappings, setAnswerMappings] = useState<AnswerMapping[]>(initialAnswerMappings || [])
+
+  const [openDeleteMappingModal, setOpenDeleteMappingModal] = useState<AnswerMapping | null>(null)
+
+
+  const deleteAnswerMapping = async () => {
+    if (!openDeleteMappingModal) {
+      return
+    }
+
+    const filteredMappings = answerMappings.filter(m => m.id !== openDeleteMappingModal.id)
+
+    onChange([], filteredMappings)
+    setAnswerMappings(filteredMappings)
+    setOpenDeleteMappingModal(null)
+  }
+
+  const [addNewAnswerMapping, setAddNewAnswerMapping] = useState(false)
+
+  const saveNewMapping = async (mapping: AnswerMapping) => {
+    const newMappings = [...answerMappings, mapping]
+
+    onChange([], newMappings)
+    setAnswerMappings(newMappings)
+    setAddNewAnswerMapping(false)
+  }
+
+  return <div className='px-3 pt-1'>
+    <p>
+      Answer mappings allow you to map answers from questions in the survey to other parts of the system.
+      For example, you could map a question which collects the participant&apos;s first name to the profile&apos;s.
+      given name field. Then, any changes the participant makes to the survey will result in changes to their profile.
+    </p>
+    <table className={'table'}>
```

Comment:
I used a regular table (rather than a `useTable`) because it was easier to achieve my desired "+" button -> "add new answer mapping" UX. I could try to implement it with a `useTable` if you all prefer.

Reply:
I got it working but I'm not sure it makes the code more readable, and it has the side effect of making the interactions a bit wonkier. I'll defer to your final judgement here.